dlt/dlt.py

```diff
@@ -822,6 +825,10 @@ def __iter__(self):  # pylint: disable=too-many-branches
                     if not index % 100000:
                         self._log_message_progress()
                     yield msg
+                else:
+                    if self.live_run and self.yield_none_allowed:
+                        logger.debug("Yield None when no new message arrives [at %s]", self.file_position)
+                        yield None
```

Comment:
The common behavior of `__iter__` is blocking mode. I would like suggest you to add another function like `non_blocking_iter` to implement this function. 

It's also a good practice to share the implementation without copying the code directly.